# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1448892958 18000
#      Mon Nov 30 09:15:58 2015 -0500
# Node ID 29861f5f28c5d23acd8b300108099823a53ad678
# Parent  38968777601b4c2512afee3671268f095cd8a696
avoid defined but not used warning in error.cc

* error.cc (debug_or_throw_exception (bool)): Delete.
(usage_1 (const char *, const char *, va_list)): Create
octave_execution_exception object and call usage_1 (const
octave_execution_exception&, const char *, const char *, va_list).

diff --git a/libinterp/corefcn/error.cc b/libinterp/corefcn/error.cc
--- a/libinterp/corefcn/error.cc
+++ b/libinterp/corefcn/error.cc
@@ -301,32 +301,16 @@ debug_or_throw_exception (const octave_e
         pr_where (std::cerr, "error");
 
       do_keyboard (octave_value_list ());
     }
   else
     throw e;
 }
 
-static void
-debug_or_throw_exception (bool show_stack_trace = false)
-{
-  octave_execution_exception e;
-
-  if (show_stack_trace && octave_exception_state != octave_exec_exception)
-    {
-      std::ostringstream buf;
-      pr_where (buf, "error");
-
-      e.set_stack_trace (buf.str ());
-    }
-
-  debug_or_throw_exception (e, show_stack_trace);
-}
-
 // Warning messages are never buffered.
 
 static void
 vwarning (const char *name, const char *id, const char *fmt, va_list args)
 {
   if (discard_warning_messages)
     return;
 
@@ -397,19 +381,19 @@ usage_1 (const octave_execution_exceptio
   verror (true, std::cerr, "usage", id, fmt, args);
 
   debug_or_throw_exception (e);
 }
 
 static void
 usage_1 (const char *id, const char *fmt, va_list args)
 {
-  verror (true, std::cerr, "usage", id, fmt, args);
+  octave_execution_exception e;
 
-  debug_or_throw_exception ();
+  usage_1 (e, id, fmt, args);
 }
 
 void
 vusage (const char *fmt, va_list args)
 {
   usage_1 ("", fmt, args);
 }
 
